File: app/src/main/java/com/dmko/pairwisecomparison/ui/screens/addeditcomparison/AddEditComparisonContract.java
package com.dmko.pairwisecomparison.ui.screens.addeditcomparison;

import com.dmko.pairwisecomparison.data.entities.Comparison;
import com.dmko.pairwisecomparison.ui.base.mvp.BasePresenter;
import com.dmko.pairwisecomparison.ui.base.mvp.BaseView;

public interface AddEditComparisonContract {

    interface View extends BaseView {

        void showLoading(boolean isLoading);

        void setComparison(Comparison comparison);

        void showEmptyNameError();

        void closeDialog();
    }

    interface Presenter extends BasePresenter<View> {

        void setArgs(String comparisonId);

        void start();

        void saveComparison(String comparisonName);
    }
}

File: app/src/main/java/com/dmko/pairwisecomparison/ui/screens/addeditcomparison/AddEditComparisonContract.kt
package com.dmko.pairwisecomparison.ui.screens.addeditcomparison

import com.dmko.pairwisecomparison.data.entities.Comparison
import com.dmko.pairwisecomparison.ui.base.mvp.BasePresenter
import com.dmko.pairwisecomparison.ui.base.mvp.BaseView

interface AddEditComparisonContract {

    interface View : BaseView {

        fun showLoading(isLoading: Boolean)

        fun setComparison(comparison: Comparison)

        fun showEmptyNameError()

        fun closeDialog()
    }

    interface Presenter : BasePresenter<View> {

        fun setArgs(comparisonId: String?)

        fun start()

        fun saveComparison(comparisonName: String)
    }
}